pkg/cmd/manager/controller.go

```diff
@@ -0,0 +1,310 @@
+	if err != nil {
+		return err, errMessage
+	}
+	if !leaderElection {
+		log.Info("Leader election is disabled!")
+	}
+
+	err, errMessage = c.ControllerManager.SetOperatorImage(bootstrapClient, ctx)
+	if err != nil {
+		return err, errMessage
+	}
+
+	options, err, errMessage := c.ControllerManager.GetManagerOptions(bootstrapClient, watchNamespace)
+	if err != nil {
+		return err, errMessage
+	}
+
+	mgr, ctrlClient, err, errMessage := c.ControllerManager.CreateManager(healthPort, monitoringPort, leaderElection, leaderElectionID, cfg, eventBroadcaster, controllerNamespace, options, ctx)
+
+	initCtx, initCancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer initCancel()
+
+	err, errMessage = c.ControllerManager.ControllerPreStartResourcesInit(initCtx, bootstrapClient, watchNamespace, controllerNamespace, ctx, ctrlClient, mgr)
+	if err != nil {
+		return err, errMessage
+	}
+
+	log.Info("Starting the manager")
+	return mgr.Start(ctx), "manager exited non-zero"
+}
+
+type BaseManager struct {
+	Log                  logutil.Logger
+	WatchNamespaceEnvVar string
+	ControllerNamespace  string
+	ControllerPodName    string
+	AddToManager         func(ctx context.Context, manager manager.Manager, client client.Client) error
+}
+
+func (bm BaseManager) LoggerSetup() (logutil.Logger, error, string) {
+	// The logger instantiated here can be changed to any logger
+	// implementing the logr.Logger interface. This logger will
+	// be propagated through the whole operator, generating
+	// uniform and structured logs.
+
+	// The constants specified here are zap specific
+	var logLevel zapcore.Level
+	logLevelVal, ok := os.LookupEnv("LOG_LEVEL")
+	if ok {
+		switch strings.ToLower(logLevelVal) {
+		case "error":
+			logLevel = zapcore.ErrorLevel
+		case "info":
+			logLevel = zapcore.InfoLevel
+		case "debug":
+			logLevel = zapcore.DebugLevel
+		default:
+			customLevel, err := strconv.Atoi(strings.ToLower(logLevelVal))
+			if err != nil {
+				return bm.Log, err, "Invalid log-level"
+			}
+			// Need to multiply by -1 to turn logr expected level into zap level
+			logLevel = zapcore.Level(int8(customLevel) * -1)
+		}
+	} else {
+		logLevel = zapcore.InfoLevel
+	}
+
+	// Use and set atomic level that all following log events are compared with
+	// in order to evaluate if a given log level on the event is enabled.
+	logf.SetLogger(zapctrl.New(func(o *zapctrl.Options) {
+		o.Development = false
+		o.Level = zap.NewAtomicLevelAt(logLevel)
+	}))
+
+	klog.SetLogger(bm.Log.AsLogger())
+
+	_, err := maxprocs.Set(maxprocs.Logger(func(f string, a ...interface{}) { bm.Log.Info(fmt.Sprintf(f, a)) }))
+
+	return bm.Log, err, "failed to set GOMAXPROCS from cgroups"
+}
+
+// getWatchNamespace returns the Namespace the operator should be watching for changes.
+func (bm BaseManager) GetWatchNamespace() (string, error) {
+	ns, found := os.LookupEnv(bm.WatchNamespaceEnvVar)
```

Comment:
looking at how the `WatchNamespaceEnvVar` is implemented, this can probably be just a function in an utility/platform package